test/stripe/stripe_client_test.rb

```diff
@@ -334,6 +334,46 @@ class StripeClientTest < Test::Unit::TestCase
         end
       end
 
+      context "API resource instance methods" do
+        should "define methods for all api resources" do
+          client = StripeClient.new
+
+          # Update Sigma name to account for nuance
+          api_resources = Stripe::Util.api_object_classes
+          sigma_class = api_resources.delete("scheduled_query_run")
+          api_resources["sigma.scheduled_query_run"] = sigma_class
+
+          api_resources.each do |string, _|
+            if string.include?(".")
+              resource_module, resource_name = string.split(".")
+
+              assert client.respond_to?(resource_module), "#{resource_module} not found"
+              assert client.send(resource_module).respond_to?("#{resource_name}s"), "#{resource_name} not found"
+            else
+              assert client.respond_to?("#{string}s"), "#{string} not found"
+            end
+          end
+        end
+
+        should "make expected request on an API resource" do
+          client = StripeClient.new(api_key: "sk_test_local")
+          account = client.accounts.retrieve("acct_1234")
+          assert_requested(:get,
+                           "#{Stripe.api_base}/v1/accounts/acct_1234",
+                           headers: { "Authorization" => "Bearer sk_test_local" })
+          assert account.is_a?(Stripe::Account)
+        end
+
+        should "make expected request on a namespace API resource" do
+          client = StripeClient.new(api_key: "sk_test_local")
+          list = client.radar.value_lists.retrieve("rsl_123")
+          assert_requested(:get,
+                           "#{Stripe.api_base}/v1/radar/value_lists/rsl_123",
+                           headers: { "Authorization" => "Bearer sk_test_local" })
+          assert list.is_a?(Stripe::Radar::ValueList)
+        end
+      end
```

Comment:
I think it might be a good idea to introduce a basic test for each of the common types of operation like list, retrieve, delete, update just for a little extra certainty that they work. (We'd need some way of better exercising our `opts` position detect code anyway.)

Would you mind also checking that passing `opts` overrides like `{api_key: 'sk_test_override'}` works as well?

Reply:
Yeah, that's a great idea and should help fill the gap that allowed the bug to slip in. I'll get those added 👍

Reply:
Excellent! TY man.

Reply:
Added some basic tests for the common operations. Just starting to crack into supporting all the other Stripe options so I imagine I'll be re-organizing these specs.